### ISSUE TITLE
Supporting various split-apply-combine spatial workflows

### ISSUE THREAD

**Issue by Aariq** [njtierney/geotargets#74]
So there are at least 4 different ways (yikes!) to do a split-apply-combine type workflow with spatial data just in `terra` that I've encountered. It may be beneficial to figure out a way to implement these workflows using `targets` branching in `geotargets`

1) SpatRasters or SpatVectors with multiple layers can be split with `[[` and recombined with `c()` or `rast()`/`vect()`.  Also can be iterated over with `lapply()` or coerced to a list with `as.list()`
2) SpatRasterCollections hold a list of SpatRasters, possibly with different extents, resolutions, and projections.  Can be split with `[` (but not `[[`) and combined with `terra::sprc()`. `lapply()` also works on them and can be coerced with `as.list()`. (see #53)
3) SpatRasterDatasets hold a list of SpatRasters with the same resolution and projection (sub-datasets).  Can be split with `[` or `[[` and re-combined with `terra::sds()`.  `lapply()` also works and can be coerced with `as.list()`.  `names` are lost in the process (I assume this is a bug: https://github.com/rspatial/terra/issues/1513). (see #59)
4) ✅ `makeTiles()` splits a raster into tiles that are saved to disk and returns a vector of file paths.  Can be opened individually, worked on, and re-combined with `merge(sprc(rast(<list of tiles as SpatRasters>)))`.  Alternatively, the files on disk can be opened with `vrt()` (see #69)
5) ✅ iterate over lists of SpatRasters or SpatVectors (e.g. with dynamic branching)

### Some open questions:

1. Which of these (if any) should we try to support better in `geotargets` (e.g. with target factories)?
2. Are there existing similar patterns we can borrow from other Targetopia packages? (knowing that we can't customize the behavior of the `iteration` argument to `tar_target_raw()`)
3. Will any of these be easier to do if we adopt an approach like #63? (I suspect that a `tar_terra_tiles()` would be easier like this)
4. Should we try to figure this stuff out soon, or later after we've added basic support for `sf` and `stars`?

**Comment by Aariq** [comment]
A 5th way: just iterating over lists of rasters or vectors (#77).  

Given that these keep coming up, and that it is confusing complicated stuff to implement (see my ongoing struggles in #76), it might be prudent to find a time we (@njtierney, I, and possibly @brownag) can talk in real-time about this.  I want to get some kind of vision about how much of this we build in to `geotargets` and how much we just document in vignettes and leave it to the users (e.g. my response in #77, or how I implemented iterating over tiles before #76). And if we do implement some solutions, how generalized can/should they be?  Would each of these 5 ways need bespoke target factories if we implemented something for them in `geotargets`?
